crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -3,6 +3,11 @@ use acvm::acir::BlackBoxFunc;
 use super::{basic_block::BasicBlockId, map::Id, types::Type, value::ValueId};
 
 /// Reference to an instruction
+///
+/// Note that InstructionIds are not unique. That is, two InstructionIds
```

Comment:
>Note that InstructionIds are not unique.

Do we mean that InstructionIds are unique but they do not refer to unique data? Meaning you can have InstructionId 5 and 6 both referring to the same data

Reply:
That is correct, yes.

This is necessary for non-pure instructions like `Instruction::Load` which may return different results if they're used twice in different spots. Since we attach instruction results to the id of the instruction, their ids must be different to allow different results.